### PR TITLE
Add CLI options to run metrics endpoint over HTTPS

### PR DESCRIPTION
**Pull request by ConnorJC3**
**Is this a bug fix or adding new feature?**

New feature (and related bugfix)

**What is this PR about? / Why do we need it?**

_[Add options to run metrics endpoint over HTTPS](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/commit/88e1b6bc4e3c5b0be93fd059b31a1e120580ee40)_

Adds the ability to run the metrics server via HTTPS by supplying a certificate and key.

_[Validate options in main.go](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/commit/1ba86c8e2461bb990ef86654501c64dc02b00cc9)_

Adds missing `Validate()` call to `main.go`

**What testing is done?** 

CI/manual/Added unit tests

```
$ AWS_REGION="us-west-2" ./bin/aws-ebs-csi-driver controller --http-endpoint=":6443" --metrics-cert-file=`pwd`/server.crt --metrics-key-file=`pwd`/server.key 
I0419 18:04:46.442557   29484 metrics.go:96] "Metric server listening" address=":6443" path="/metrics"
I0419 18:04:46.444976   29484 driver.go:68] "Driver Information" Driver="ebs.csi.aws.com" Version="v1.29.1"

$ curl -kv https://localhost:6443
*   Trying 127.0.0.1:6443...
* Connected to localhost (127.0.0.1) port 6443
* ALPN: curl offers h2,http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=XX; L=Default City; O=Default Company Ltd
*  start date: Apr 19 18:02:41 2024 GMT
*  expire date: Apr 17 18:02:41 2034 GMT
*  issuer: C=XX; L=Default City; O=Default Company Ltd
*  SSL certificate verify result: self signed certificate (18), continuing anyway.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://localhost:6443/
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: localhost:6443]
* [HTTP/2] [1] [:path: /]
* [HTTP/2] [1] [user-agent: curl/8.3.0]
* [HTTP/2] [1] [accept: */*]
> GET / HTTP/2
> Host: localhost:6443
> User-Agent: curl/8.3.0
> Accept: */*
> 
< HTTP/2 404 
< content-type: text/plain; charset=utf-8
< x-content-type-options: nosniff
< content-length: 19
< date: Fri, 19 Apr 2024 18:05:05 GMT
< 
404 page not found
```

### CODE REVIEW

File: pkg/driver/options.go [hunk 1]
```diff
@@ -83,6 +87,8 @@ func (o *Options) AddFlags(f *flag.FlagSet) {
 	// Server options
 	f.StringVar(&o.Endpoint, "endpoint", DefaultCSIEndpoint, "Endpoint for the CSI driver server")
 	f.StringVar(&o.HttpEndpoint, "http-endpoint", "", "The TCP network address where the HTTP server for metrics will listen (example: `:8080`). The default is empty string, which means the server is disabled.")
+	f.StringVar(&o.MetricsCertFile, "metrics-cert-file", "", "The path to a certificate to use for serving the metrics server over HTTPS. If the certificate is signed by a certificate authority, this file should be the concatenation of the server's certificate, any intermediates, and the CA's certificate. If this is non-empty, --http-endpoint and --metrics-key-file MUST also be non-empty.")
```
Comment (AndrewSirenko) [inline pkg/driver/options.go:90]: Add `MetricsCertFile` and `MetricsKeyFile` to [aws-ebs-csi-driver/docs/options.md at master · kubernetes-sigs/aws-ebs-csi-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/options.md)

**Review by AndrewSirenko — CHANGES_REQUESTED** [review]
Code itself lgtm other than missing options documentation page updates

Thank you for removing the 2 panics.

File: pkg/driver/options.go (hunk 1 above)
Comment (ConnorJC3) [reply pkg/driver/options.go:90]: Done

**Review by AndrewSirenko — APPROVED** [review]

**Review by torredil — APPROVED** [review]
/approve

**Pull request merged**